void-docs.in

```diff
@@ -0,0 +1,145 @@
+#!/bin/sh
+# This is the void linux documentation browser
+
+[ "$DEBUG" ] && set -x
+# turn on errors and noglob
+set -fe
+
+# document to open
+VOIDDOC=
+# terms to search for
+TERMS=
+# versions to try
+_html=1
+_md=1
+_roff=1
+
+usage() {
+    cat <<EOF
+Welcome to the Void Linux documentation browser!
+
+Usage: $0 [options] [search terms]
+
+Options:
+-h, --help: show this help text
+-b: only try to display the HTML version
+-m: only try to display the Markdown version
+-r: only try to display the roff (mandoc) version
+-s: only display search results
+EOF
+
+    exit 0
+}
+
+while [ $# -gt 0 ]; do
+    case $1 in
+        -h|--help) usage ;;
+        -b) _html=1 ; _md= ; _roff= ;;
+        -m) _html= ; _md=1 ; _roff= ;;
+        -r) _html= ; _md= ; _roff=1 ;;
+        -s) _search=1 ;;
+        *)
+            if [ "$TERMS" ]; then
+                TERMS="$TERMS -and -iwholename *$1*"
+            else
+                TERMS="-iwholename *$1*"
+            fi
+            ;;
+    esac
+    shift
+done
+
+: "${DOCS:=@PREFIX@/share/doc/void}"
+
+# creates the appropriate path
+find_page() {
+    case $1 in
+        html) VOIDDOC="$DOCS/html/$2.html" ;;
+        md) VOIDDOC="$DOCS/markdown/$2.md" ;;
+        man) VOIDDOC="$DOCS/mandoc/$2.7" ;;
+    esac
+
+    if [ -f "$VOIDDOC" ]; then
+        echo "$VOIDDOC"
+    else
+        echo "couldn't find file" >&2
+        return 1
+    fi
+}
+
+# knows how to open a certain page
+open_page() {
+    if [ "$_html" ]; then
+        # check if wayland or X session
+        if [ -n "$WAYLAND_DISPLAY" -o -n "$DISPLAY" ] && command -v xdg-open >/dev/null; then
+            page=$(find_page html $1)
+            exec xdg-open $page
+        fi
+
+        if [ -n "$BROWSER" ] && command -v $BROWSER >/dev/null; then
```

Comment:
I think `$BROWSER` should be given more priority than `xdg-open`.
`xdg-open` could be pulled by some packages,
`$BROWSER` is usually set explicitly by users.

Reply:
Good idea, thanks.